kalite/distributed/management/commands/screenshots.py

```diff
@@ -256,14 +257,21 @@ def snap(self, slug, focus, note):
                  (self.browser.current_url, self.browser.title, slug, settings.SCREENSHOTS_EXTENSION))
 
         if focus:
-            # Apply the specified styles to element. Currently only selection by
-            # id is supported. TODO: Extend it a more generic CSS selector.
+            self.browser_wait_for_js_object_exists("$")
             selector = focus['selector']
             styles = focus['styles']
-            for key, value in styles.iteritems():
-                self.browser.execute_script('$("%s").css("%s", "%s");' % (selector, key, value))
-            if note:
-                self.browser.execute_script("$('%s').qtip({content:{text:\"%s\"},show:{ready:true,delay:0,effect:false}})" % (selector, note))
+            try:
+                for key, value in styles.iteritems():
+                    self.browser.execute_script('$("%s").css("%s", "%s");' % (selector, key, value))
+                if note:
+                    self.browser.execute_script("$('%s').qtip({content:{text:\"%s\"},show:{ready:true,delay:0,effect:false}})" % (selector, note))
+            except WebDriverException as e:
+                log.error("Error taking screenshot:")
+                log.error(str(e))
+                log.error("Screenshot info: {0}".format((focus, note)))
+                log.error("Current url: {0}".format(self.browser.current_url))
+                import sys
+                sys.exit(1)
```

Comment:
Because I want the process to exit with a non-zero code, so that the parent process (building the docs) recognizes something amiss. Simply raising an exception doesn't appear to give a nonzero exit code.